components/brave_wallet/browser/keyring_service.cc

```diff
@@ -807,16 +813,43 @@ const std::string KeyringService::GetMnemonicForKeyringImpl(
   return std::string(mnemonic.begin(), mnemonic.end());
 }
 
+void KeyringService::AddFilecoinAccount(const std::string& account_name,
+                                        const std::string& network,
+                                        AddAccountCallback callback) {
+  if (!IsFilecoinEnabled()) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  std::string keyring_id = GetFilecoinKeyringId(network);
+
+  if (!LazilyCreateKeyring(keyring_id)) {
+    VLOG(1) << "Unable to create Filecoin keyring";
+    std::move(callback).Run(false);
+    return;
+  }
+
+  auto* keyring = GetHDKeyringById(keyring_id);
+
+  if (keyring) {
+    AddAccountForKeyring(keyring_id, account_name);
+  }
+
+  NotifyAccountsChanged();
+
+  std::move(callback).Run(keyring);
+}
+
 void KeyringService::AddAccount(const std::string& account_name,
                                 mojom::CoinType coin,
                                 AddAccountCallback callback) {
-  std::string keyring_id = GetKeyringIdForCoin(coin);
-  if (keyring_id == mojom::kFilecoinKeyringId) {
+  std::string keyring_id = GetMainKeyringIdForCoin(coin);
+  if (IsFilecoinKeyringId(keyring_id)) {
```

Comment:
We should block Filecoin here because it needs to be handled in `AddFilecoinAccount`